libvncclient/tls_gnutls.c

```diff
@@ -347,11 +348,16 @@ ReadVeNCryptSecurityType(rfbClient* client, uint32_t *result)
             t==rfbVeNCryptX509VNC ||
             t==rfbVeNCryptX509Plain)
         {
-            flag++;
-            authScheme=t;
-            rfbClientLog("Selecting security type %d (%d/%d in the list)\n", authScheme, loop, count);
-            /* send back 4 bytes (in original byte order!) indicating which security type to use */
-            if (!WriteToRFBServer(client, (char *)&tAuth[loop], 4)) return FALSE;
+            if (
```

Comment:
Very good modeling of the preferring of encrypted schemes while keeping the "pick first, list rest" semantics :clap: !